### PR TITLE
feat: add pagination to ListResources API

### PR DESCRIPTION
**Pull request by FemiNoviaLina**
[Shield] Add pagination to ListResources API

### CODE REVIEW

File: internal/store/postgres/resource_repository.go [hunk 1]
```diff
@@ -84,6 +84,17 @@ func (r ResourceRepository) Create(ctx context.Context, res resource.Resource) (
 func (r ResourceRepository) List(ctx context.Context, flt resource.Filter) ([]resource.Resource, error) {
 	var fetchedResources []Resource
 
+	var defaultLimit int32 = 50
```
Comment (mabdh) [inline internal/store/postgres/resource_repository.go:87]: Lets add more test cases to validate this pagination

Comment (FemiNoviaLina) [reply internal/store/postgres/resource_repository.go:87]: Sure, I am also thinking that the test should cover more cases.

Comment (FemiNoviaLina) [reply internal/store/postgres/resource_repository.go:87]: added these cases:
- given page number >= 1, should return resources in given page
- given page number <=0, should return resources in page 1
- given limit <= 0, should return max 50 resources
- should return filtered resources in given page number
- should return empty resource if no more resources found in page

File: internal/api/v1beta1/resource.go [hunk 2]
```diff
@@ -36,14 +36,17 @@ func (h Handler) ListResources(ctx context.Context, request *shieldv1beta1.ListR
...
 
-	resourcesList, err := h.resourceService.List(ctx, filters)
+	resourcesResp, err := h.resourceService.List(ctx, filters)
 	if err != nil {
 		logger.Error(err.Error())
 		return nil, grpcInternalServerError
 	}
 
+	resourcesList := resourcesResp.Resources
 	for _, r := range resourcesList {
```
Comment (mabdh) [inline internal/api/v1beta1/resource.go]:
```suggestion
	for _, r := range resourcesResp.Resources {
```

**Review by mabdh — APPROVED** [review]

**Pull request merged**